pcap/get_pcap_handle_linux.go

```diff
@@ -0,0 +1,56 @@
+//go:build linux
+
+package pcap
+
+import (
+	"time"
+
+	"github.com/akitasoftware/go-utils/optionals"
+	"github.com/containernetworking/plugins/pkg/ns"
+	"github.com/google/gopacket/pcap"
+	"github.com/pkg/errors"
+)
+
+func GetPcapHandle(
+	interfaceName string,
+	snaplen int32,
+	promisc bool,
+	timeout time.Duration,
+	targetNetworkNamespaceOpt optionals.Optional[string],
+) (*pcap.Handle, error) {
+	var (
+		handle *pcap.Handle
+		err    error
+	)
+
+	if targetNetworkNamespace, exists := targetNetworkNamespaceOpt.Get(); exists {
+		// Switch to the target network namespace.
+		targetNs, err := ns.GetNS(targetNetworkNamespace)
+		if err != nil {
+			return nil, errors.Wrapf(err, "can't get network namespace %s", targetNetworkNamespace)
+		}
+		//TODO(K8s-MNS) Is this right place to close? Same in net.go
+		defer targetNs.Close()
```

Comment:
I think this should be fine, it's just a file handle underneath so it doesn't need to be closed at a particular point in time, any time after we use it should be OK.